core/src/main/java/tc/oc/pgm/projectile/ProjectileDefinition.java

```diff
@@ -50,6 +56,31 @@ public ProjectileDefinition(
     this.throwable = throwable;
     this.precise = precise;
     this.blockMaterial = blockMaterial;
+    this.scale = scale;
+    this.solidBlockCollision = solidBlockCollision;
+    this.maxTravelTime = maxTravelTime;
+  }
+
+  public static sealed class ProjectileEntity {
```

Comment:
A class with no members and no functions, may as well be a sealed interface instead:
```suggestion
  public sealed interface ProjectileEntity permits RealEntity, CustomEntity {
```